bundles/binding/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/ADZoneTracker.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2010-2016, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.alarmdecoder.internal;
+
+import org.openhab.binding.alarmdecoder.internal.model.ADZone;
+import java.util.ArrayList;
+import java.util.ListIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Track zone faults / restores by zone ID watching the alarm panel
+ * message stream
+ *
+ * @author Sean Mathews <coder@f34r.com>
+ * @since 1.6.0
+ */
+public class ADZoneTracker {
+
+    private static final Logger logger = LoggerFactory.getLogger(ADZoneTracker.class);
+
+    private ArrayList<ADZone> _zoneList;
+    private boolean _lastReady;
+    private int _lastZone = 0;
+
+    // TODO: zone timeout handling. zone list garbage collector.
+    long _currTime = 0;
+
+
+    ADZoneTracker() {
+        _zoneList = new ArrayList<ADZone>();
+        _lastReady = false;
+        _currTime = System.nanoTime();
+    }
+
+    /**
+     * get the state of a zone
+     *
+     * @param Zone - the zone
+     * @return return zone status or null if the zone is invalid
+     *     -1 = unknown
+     *      0 = restored
+     *      1 = faulted
+     */
+    public boolean getZoneState(int Zone) {
```

Comment:
Only use ProperCase for class and interface names, so parameter would be `zone`.